pkg/sql/crdb_internal.go

```diff
@@ -1133,11 +1134,18 @@ CREATE TABLE crdb_internal.jobs (
   num_runs              INT,
   execution_errors      STRING[],
   execution_events      JSONB,
+  INDEX(job_id),
```

Comment:
Can you add a test that this is having the desired effect, either by outputting a trace, or the explain for `SHOW JOB`?

Reply:
I added an explain and a trace for the top level `SHOW JOB` behaviour. The trace is skipped in our tenant test configs only because the trace has tenant-prefixed keys in that case.

Reply:
the testing framework that we use to verify changes like this have their desired effect are in the rttanalysis package. it lets you verify the number of KV lookups that are happening when a query is executed. See https://github.com/cockroachdb/cockroach/pull/111819 and https://github.com/cockroachdb/cockroach/pull/104547/ for examples.

Reply:
useful commands:
```
# to rewrite the "expected number of round trips" file:
./dev test --rewrite pkg/bench/rttanalysis -f=TestBenchmarkExpectation/ --test-args '-rewrite-iterations=10'

# to compare benchmark improvements between two commits (replace BenchmarkORMQueries/django with your test case name):
N=10 BENCHTIMEOUT=24h PKG=./pkg/bench/rttanalysis/ BENCHES=BenchmarkORMQueries/django ./scripts/bench 'HEAD' 'HEAD^'
```

Reply:
@rafiss Thanks! I'll move my tests there.